go.mod

```diff
@@ -21,6 +21,7 @@ require (
 	cloud.google.com/go/pubsub v1.17.1
 	cloud.google.com/go/storage v1.18.2
 	cloud.google.com/go/workflows v1.0.0
+	github.com/Azure/azure-event-hubs-go v1.3.1
 	github.com/Azure/azure-event-hubs-go/v3 v3.3.16
```

Comment:
You must have an import somewhere that is missing `/v3/`, that's why it pulled this old version.